jax/_src/dispatch.py

```diff
@@ -228,7 +228,7 @@ def lower_xla_callable(fun: lu.WrappedFun, device, backend, name,
   backend = xb.get_device_backend(device) if device else xb.get_backend(backend)
 
   if (config.jax_dynamic_shapes and jaxpr_has_bints(jaxpr) and
-      backend.platform != 'iree'):
+      not backend.platform == 'iree'):
```

Comment:
Maybe give this a more declarative name via, e.g.

```
def _backend_supports_unbounded_dynamic_shapes(b):
 ...
```

(Probably even better to make it a property on the backend, but not today.)